src/Sarif.Viewer.VisualStudio/Fixes/EditActionPreviewProvider.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Copyright (c) Microsoft. All rights reserved. 
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+
+namespace Microsoft.Sarif.Viewer.Fixes
+{
+    using System;
+    using System.Collections.Generic;
+    using System.ComponentModel.Composition;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using System.Windows;
+    using Microsoft.VisualStudio.Text;
+    using Microsoft.VisualStudio.Text.Differencing;
+    using Microsoft.VisualStudio.Text.Editor;
+    using Microsoft.VisualStudio.Text.Projection;
+
+    [Export(typeof(IPreviewProvider))]
+    public class EditActionPreviewProvider : IPreviewProvider
```

Comment:
>EditActionPreviewProvider [](start = 17, length = 25)

Borrowed code, except that we had to change `[Export]` to the more explicit `[Export(typeof(IPreviewProvider)]` because (presumably) we're using an older version of MEF.